QPM/SymLinker.Linker/Linker.cs

```diff
@@ -0,0 +1,269 @@
+using System.Runtime.InteropServices;
+using System.Text;
+
+// Stolen from https://github.com/huffSamuel/SymLinker
+// Cleaned up by Fern, because the old code was horrible
+namespace SymLinker.Linker
+{
+    public class Linker
+    {
+        #region Actions
+        // Attach to these to get warnings, errors, and informaion out of SymLinkerCore
+        public Action<string> OnWarn;
+        public Action<string> OnError;
+        public Action<string> OnInfo;
+        #endregion
+
+        private List<Error> errors = new List<Error>();
+        private ISymLinkCreator linker = null;
+
+        /// <summary>
+        /// Creates a new Symbolic Link Creator
+        /// </summary>
+        public Linker()
+        {
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                linker = new WindowsSymLinkCreator();
+            }
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            {
+                linker = new LinuxSymLinkCreator();
+            }
+            else
+            {
+                linker = new OSXSymLinkCreator();
+            }
+        }
+
+        /// <summary>
+        /// Creates a symbolic link from <paramref name="source"/> to <paramref name="dest"/>
+        /// </summary>
+        /// <param name="source">Source file</param>
+        /// <param name="dest">Destination directory</param>
+        /// <returns>
+        /// Returns true if the system was able to create the SymLink
+        /// </returns>
+        public void CreateLink(string source, string dest)
+        {
+            if (!CheckLinkReadiness(source, dest))
+            {
+                WriteError(ResolveError());
+            }
+            else
+            {
+                var linkMade = linker.CreateSymLink(source, dest, true);
+
+                if (!linkMade)
+                    throw new IOException("Failed to create link");
+            }
+        }
+
+        /// <summary>
+        /// Checks for readiness of a drive to perform a SymLink creation. Expects absolute path
+        /// </summary>
+        /// <param name="source">Source file path</param>
+        /// <param name="dest">Destination directory path</param>
+        /// <returns>
+        /// Returns true if the system is ready to perform a SymLink
+        /// </returns>
+        private bool CheckLinkReadiness(string source, string dest)
+        {
+            try
+            {
+                var destDrive = dest.Substring(0, 1)[0];
+                var fileSize = 0L;
+                var freeSpace = 0L;
+
+                // Check format
+                if (!char.IsLetter(destDrive))
+                    errors.Add(Error.DestinationNotAbsolutePath);
+
+                // Check existance
+                if (File.Exists(source))
+                {
+                    fileSize = new FileInfo(source).Length;
+                    if (fileSize < 1024) WriteWarn("File less than standard block size");
```

Comment:
Magic number, make `private const` instead, and I even think there's a constant for that somewhere in `System.IO`? maybe something like `BlockSize`, idk